cate/ds/esa_cci_odp.py

```diff
@@ -304,21 +338,23 @@ async def _fetch_opensearch_feature_list(base_url, query_args, monitor: Monitor
     full_feature_list = []
     with monitor.starting("Loading", 10):
         async with aiohttp.ClientSession() as session:
-            # todo remove this when the opensearch server provides results of more than 10,000 features
-            while start_page < 11:
+            while True:
                 monitor.progress(work=1)
                 paging_query_args = dict(query_args or {})
                 paging_query_args.update(startPage=start_page, maximumRecords=maximum_records,
                                          httpAccept='application/geo+json')
                 url = base_url + '?' + urllib.parse.urlencode(paging_query_args)
-                resp = await session.request(method='GET', url=url)
-                resp.raise_for_status()
-                json_text = await resp.read()
-                json_dict = json.loads(json_text.decode('utf-8'))
-                feature_list = json_dict.get("features", [])
-                full_feature_list.extend(feature_list)
-                if len(feature_list) < maximum_records:
-                    break
+                resp = await get_response(session, url)
+                if resp:
+                    json_text = await resp.read()
+                    try:
+                        json_dict = json.loads(json_text.decode('utf-8'))
+                        feature_list = json_dict.get("features", [])
+                        full_feature_list.extend(feature_list)
+                        if len(feature_list) < maximum_records:
+                            break
+                    except json.decoder.JSONDecodeError:
+                        print(json_text)
```

Comment:
Again, missing error handling. Is this ok, because the JSON's content is not so important?